c/tskit/trees.c

```diff
@@ -1232,6 +1232,26 @@ tsk_treeseq_check_windows(const tsk_treeseq_t *self, tsk_size_t num_windows,
     return ret;
 }
 
+static int
+check_time_windows(tsk_size_t num_time_windows, const double *time_windows)
+{
+    int ret = TSK_ERR_BAD_TIME_WINDOWS;
+    tsk_size_t j;
+
+    if (num_time_windows == 1) {
+        ret = TSK_ERR_BAD_NUM_TIME_WINDOWS;
+        goto out;
```

Comment:
We would want to test this in C. These are error conditions that you can imagine, e.g., we don't correctly clean memory up after. So, it's important to provoke them when running in valgrind.